language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1895,4 +1888,19 @@ public static String getModifiedUri(WorkspaceManager workspaceManager, String ur
         }
     }
 
+    /**
+     * Given a node (currentNode) start looking up the parent ladder until the given predicate is satisfied.
+     *
+     * @param currentNode Node to start looking
+     * @param predicate   to be satisfied
+     * @return {@link Optional}
+     */
+    public static Optional<Node> getMatchingNode(Node currentNode, Predicate<Node> predicate) {
+        Node evalNode = currentNode;
```

Comment:
In this variable necessary?

Reply:
Created this as a temp variable in order to avoid updating the method parameter

Reply:
But since we're not using this variable inside the method, do we need to worry about that?

Reply:
Basically it is better if we are not going to alter the parameter value, unless it is necessary